pretyping/typing.ml

```diff
@@ -110,33 +97,53 @@ let judge_of_apply env sigma funj argjv =
   in
   apply_rec sigma 1 (Esubst.subs_id 0) funj.uj_type (Array.to_list argjv)
 
+let judge_of_applied ~check env sigma funj argjv =
+  let sigma =
+    if check then
+      let (sigma, _) = judge_of_apply_core env sigma funj argjv in
+      sigma
+    else sigma
+  in
+  let typ = hnf_prod_appvect env sigma (j_type funj) (Array.map j_val argjv) in
+  sigma, { uj_val = (mkApp (j_val funj, Array.map j_val argjv)); uj_type = typ }
```

Comment:
Why do we do the hnf_prod_appvect in the check = true case instead of getting the result of judge_of_apply_core?

Reply:
No idea, this is just the straightforward port of the previous state of the code.